packages/salesforcedx-vscode-apex/src/languageUtils/languageServerUtils.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { execSync } from 'child_process';
+import { SIGKILL } from 'constants';
+import { UBER_JAR_NAME } from '../constants';
+import { telemetryService } from '../telemetry';
+
+export type ProcessDetail = {
+  pid: number;
+  ppid: number;
+  command: string;
+  orphaned: boolean;
+};
+
+export function findAndCheckOrphanedProcesses(): ProcessDetail[] {
+  const platform = process.platform.toLowerCase();
+  const isWindows = platform === 'win32';
+
+  const cmd = isWindows
+    ? `Get-CimInstance -ClassName Win32_Process | Where-Object { $_.CommandLine -like '*${UBER_JAR_NAME}*' } | ForEach-Object { [PSCustomObject]@{ ProcessId = $_.ProcessId; ParentProcessId = $_.ParentProcessId; CommandLine = $_.CommandLine } } | Format-Table -HideTableHeaders`
+    : `ps -e -o pid,ppid,command | grep "${UBER_JAR_NAME}"`;
+
+  const stdout = execSync(
+    isWindows ? `powershell.exe -command "${cmd}"` : cmd
+  ).toString();
```

Comment:
In other parts of the extension we use cross_spawn for executing commands.  Think we should stick with that? I'm not overly familiar with windows and don't know if all systems will have powershell.exe or have it in the path. See [here](https://github.com/forcedotcom/salesforcedx-vscode/blob/69edd89386d8287230b5230423d9e3b44a32a981/packages/salesforcedx-utils/src/cli/cliCommandExecutor.ts#L67)

Reply:
We could also consider using something like ps-node https://www.npmjs.com/package/ps-node

Reply:
I will look into both
Powershell seemd to provide more concise solution.
I will reexamine solving the process detection and verification with cmd.